src/elf/dyn.rs

```diff
@@ -547,6 +515,23 @@ macro_rules! elf_dyn_std_impl {
                 needed
             }
         }
+    };
+}
+
+macro_rules! elf_dynamic_info_std_impl {
+    ($size:ident, $phdr:ty) => {
+        /// Convert a virtual memory address to a file offset
+        fn vm_to_offset(phdrs: &[$phdr], address: $size) -> Option<$size> {
```

Comment:
I have some reservations about looping over the phdrs everytime an address lookup is requested; we really need to create an interval tree (and we could also use this in PE, which has very similar lookups).

I know you're trying to avoid allocation, but core dumps, for example, can have hundreds of phdrs, but then that might not ever have any bias calculations? I guess if this only occurs in the single pass over the dynamic info we're ok. I dunno, just some thoughts here, as opposed to allocating an associative list or a proper interval tree (which rust afaics doesn't have a robust one with a nice api unfortunately...)

Reply:
I think it would be overkill to do that for the single pass over the dynamic info.